content/docs/reference/cli-branches.md

```diff
@@ -0,0 +1,198 @@
+
+```bash
+neonctl branches create --project.id spring-sky-578180
+┌─────────────────────────┬─────────────────────────┬──────────────────────┐
+│ Id                      │ Name                    │ Created At           │
+├─────────────────────────┼─────────────────────────┼──────────────────────┤
+│ br-withered-king-763176 │ br-withered-king-763176 │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────────────────┴──────────────────────┘
+```
+
+### update
+
+This subcommand allows you to update a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches update [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `update` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :-----: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.name |             | string |         |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+```bash
+neonctl branches update --project.id spring-sky-578180 --branch.id br-withered-king-763176 --branch.name mynewbranch
+┌─────────────────────────┬─────────────┬──────────────────────┐
+│ Id                      │ Name        │ Created At           │
+├─────────────────────────┼─────────────┼──────────────────────┤
+│ br-withered-king-763176 │ mynewbranch │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────┴──────────────────────┘
+```
+
+### delete
+
+This subcommand allows you to delete a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches delete [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `delete` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :------: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+<CodeBlock shouldWrap>
+
+```bash
+neonctl branches delete --project.id spring-sky-578180 --branch.id br-withered-king-763176
+┌─────────────────────────┬─────────────┬──────────────────────┐
+│ Id                      │ Name        │ Created At           │
+├─────────────────────────┼─────────────┼──────────────────────┤
+│ br-withered-king-763176 │ mynewbranch │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────┴──────────────────────┘
+```
+
+</CodeBlock>
+
+### get
+
+This subcommand allows you to retrieve details about a branch.
+
+#### Usage
+
+```bash
+neonctl branches get [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `get` subcommand supports this option:
+
+#### Options
+
+| Option        | Description | Type   | Required |
+| ------------- | ----------- | ------ | :------: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.id   | Branch ID   | string | &check; |
```

Comment:
@duskpoet , can we also add the support of querying branches by name?
we have this in the API too, this is much more friendly for the users IMO

Reply:
I believe we have address this. You can now specify the branch name or id with get command